packages/ws-worker/src/api/execute.ts

```diff
@@ -215,34 +263,41 @@ export function onWorkflowStart(
   return sendEvent<ATTEMPT_START_PAYLOAD>(channel, ATTEMPT_START);
 }
 
+// TODO what this needs to do is look at all the job states
+// find the higher priority
+// And return that as the highest exit reason
 export async function onWorkflowComplete(
   { state, channel, onComplete }: Context,
   _event: WorkflowCompleteEvent
 ) {
   const result = state.dataclips[state.lastDataclipId!];
-
+  const reason = calculateAttemptExitReason(state);
   await sendEvent<ATTEMPT_COMPLETE_PAYLOAD>(channel, ATTEMPT_COMPLETE, {
```

Comment:
Here's the diff on the `attempt:complete` payload. `status` is removed, and `reason` is as `run:complete`